cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -399,6 +533,44 @@ struct ISOCalendar {
   }
 };
 
+template <template <typename...> class Op, typename OutType>
+std::shared_ptr<ScalarFunction> MakeTemporalZoned(std::string name,
```

Comment:
I'm sure you can find a way to share this with `MakeTemporal` below.

Reply:
Done.